src/components/transaction-list/TransactionList.js

```diff
@@ -83,8 +89,8 @@ class TransactionList extends React.PureComponent {
     const addressOrEns = accountENS || this.formatAddress(accountAddress);
 
     const data = {
-      requests,
-      transactions,
+      requests: requests.map(this.checksumAddress),
```

Comment:
I don't think you need this for requests

Reply:
also @jinchung , can you confirm if we can checksum the address at the transaction parser level and prevent this additional loop?

Reply:
If it is possible, I'd love to see it done on the backend side so we don't have to apply any processing to the data. Moreover, would be great to compute a transaction state there as well. As of now, we have to operate transaction.status and transaction.type in order to derive the actual transaction state. We implement this logic on both front-end and back-end and it seems unnecessary complex. Anyway, let me know what you think

Reply:
Sure - I'll take care of these two things: checksumAddress and transaction state

Reply:
@brunobar79 If all else looks good to you, you can merge this into Christian's branch and I'll create a separate ticket / PR